clientconn.go

```diff
@@ -98,8 +98,10 @@ type dialOptions struct {
 	// This is to support v1 balancer.
 	balancerBuilder balancer.Builder
 	// This is to support grpclb.
-	resolverBuilder  resolver.Builder
-	waitForHandshake bool
+	resolverBuilder resolver.Builder
+	// Custom user options for resolver.Build.
+	resolverBuildUserOptions interface{}
```

Comment:
nit: maybe buildResolverUserOptions? or userOptionsForBuildResolver?

Reply:
resolverBuilderUserOptions?

I'm also OK with this as-is.

Reply:
The idea was this is like `resolver.Build(UserOptions)`.
Keeping this as it is now.